state-transition/core/interfaces.go

```diff
@@ -49,6 +50,10 @@ type ReadOnlyContext interface {
 
 // ExecutionEngine is the interface for the execution engine.
 type ExecutionEngine interface {
+	NotifyForkchoiceUpdate( // added to simplify mocks
```

Comment:
I am reusing the mock coming from this transactions in `blockchain` package, which requires `NotifyForkchoiceUpdate` method too.
This is counter to what we decided so far about interfaces, but doing it just to reduce Loc changed. Happy to have a different solution once I have written all the tests I care to have